src/mpid/common/shm/mpidu_shm_alloc.c

```diff
@@ -311,15 +212,16 @@ int MPIDU_shm_seg_commit(MPIDU_shm_seg_t * memory, MPIDU_shm_barrier_t ** barrie
     }
     while (!ALLOCQ_EMPTY());
 
-    mpi_errno = check_alloc(memory, *barrier, num_local, local_rank);
+    mpi_errno = check_alloc(memory, num_local, local_rank);
     MPIR_ERR_CHECK(mpi_errno);
 
+    /* for now return null pointer for barrier */
+    *barrier = NULL;
     MPIR_CHKPMEM_COMMIT();
   fn_exit:
     /* reset segment_len to zero */
     segment_len = 0;
```

Comment:
`segment_len` is a local variable, why reset it?

Reply:
This is a leftover after the commit cleanup. I will remove it